integration/docker/volume_test.go

```diff
@@ -90,4 +96,58 @@ var _ = Describe("docker volume", func() {
 			Expect(stdout).NotTo(ContainSubstring(testFile))
 		})
 	})
+
+	Context("creating a text file under /dev", func() {
+		It("should be passed the content to the container", func() {
+			if os.Getuid() != 0 {
+				Skip("only root user can create files under /dev")
+			}
+			fileName := "/dev/foo"
+			textContent := "hello"
+			err := ioutil.WriteFile(fileName, []byte(textContent), 0644)
+			Expect(err).ToNot(HaveOccurred())
+			defer os.Remove(fileName)
+
+			args = []string{"--name", id, "-v", fileName + ":" + fileName, Image, "cat", fileName}
+			stdout, _, exitCode = dockerRun(args...)
+			Expect(exitCode).To(Equal(0))
+			Expect(stdout).To(ContainSubstring(textContent))
+
+			Expect(RemoveDockerContainer(id)).To(BeTrue())
+			Expect(ExistDockerContainer(id)).NotTo(BeTrue())
+		})
+	})
+
+	Context("passing a block device", func() {
+		It("should be mounted", func() {
+			if os.Getuid() != 0 {
+				Skip("only root user can create loop devices")
+			}
+
+			for i := 0; i < loopDevices; i++ {
+				diskFile, loopFile, err = createLoopDevice()
+				Expect(err).ToNot(HaveOccurred())
+				diskFiles = append(diskFiles, diskFile)
+				loopFiles = append(loopFiles, loopFile)
+			}
+
+			args = []string{"--name", id, "--cap-add=SYS_ADMIN", "-v", loopFile + ":" + loopFile, DebianImage, "mount", loopFile, "/mnt"}
+			stdout, _, exitCode = dockerRun(args...)
+			Expect(exitCode).To(Equal(0))
+			Expect(stdout).To(ContainSubstring(loopFile))
+
+			for _, lf := range loopFiles {
+				err = deleteLoopDevice(lf)
+				Expect(err).ToNot(HaveOccurred())
+			}
+
+			for _, df := range diskFiles {
+				err = os.Remove(df)
+				Expect(err).ToNot(HaveOccurred())
+			}
```

Comment:
you don't need two arrays (`loopFiles` and `diskFiles`) for 1 loop device

Reply:
`err = deleteLoopDevice(loopFile)` and `err = os.Remove(diskFile)`  should be enough